# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/nsRefreshDriver.h
# Commit: c1e64e8e4665
# Full Hash: c1e64e8e466506d1d9a36d747df25527bd545b10
# Author: Martin Robinson <mrobinson@igalia.com>
# Date: 2022-12-01 16:18:29
# Regressor Bug: 1791759
# File Overlap Count: 1
# Description:
#   Bug 1791759 - Add support for `content-visibility: auto` r=emilio
#   
#   This change adds support for `content-visibilty: auto` as well as
#   showing and hiding content based on the relevancy of the content as
#   defined in the specification. Changes to relevancy are handled by
# ==============================================================================

diff -r 6d4d56f57b7b -r c1e64e8e4665 layout/base/nsRefreshDriver.h
--- a/layout/base/nsRefreshDriver.h	Thu Dec 01 01:23:08 2022 +0200
+++ b/layout/base/nsRefreshDriver.h	Wed Nov 30 18:00:27 2022 +0000
@@ -417,6 +417,11 @@
     mNeedToUpdateIntersectionObservations = true;
   }
 
+  void EnsureContentRelevancyUpdateHappens() {
+    EnsureTimerStarted();
+    mNeedToUpdateContentRelevancy = true;
+  }
+
   // Register a composition payload that will be forwarded to the layer manager
   // if the current or upcoming refresh tick does a paint.
   // If no paint happens, the payload is discarded.
@@ -429,9 +434,10 @@
     eHasObservers = 1 << 0,
     eHasImageRequests = 1 << 1,
     eNeedsToUpdateIntersectionObservations = 1 << 2,
-    eHasVisualViewportResizeEvents = 1 << 3,
-    eHasScrollEvents = 1 << 4,
-    eHasVisualViewportScrollEvents = 1 << 5,
+    eNeedsToUpdateContentRelevancy = 1 << 3,
+    eHasVisualViewportResizeEvents = 1 << 4,
+    eHasScrollEvents = 1 << 5,
+    eHasVisualViewportScrollEvents = 1 << 6,
   };
 
   void AddForceNotifyContentfulPaintPresContext(nsPresContext* aPresContext);
@@ -473,6 +479,7 @@
   MOZ_CAN_RUN_SCRIPT
   void RunFrameRequestCallbacks(mozilla::TimeStamp aNowTime);
   void UpdateIntersectionObservations(mozilla::TimeStamp aNowTime);
+  void UpdateRelevancyOfContentVisibilityAutoFrames();
 
   enum class IsExtraTick {
     No,
@@ -611,6 +618,10 @@
   // all our documents.
   bool mNeedToUpdateIntersectionObservations : 1;
 
+  // True if we need to update the relevancy of `content-visibility: auto`
+  // elements in our documents.
+  bool mNeedToUpdateContentRelevancy : 1;
+
   // True if we're currently within the scope of Tick() handling a normal
   // (timer-driven) tick.
   bool mInNormalTick : 1;